handler/src/main/java/io/netty/handler/ssl/ReleasableOpenSslServerContext.java

```diff
@@ -0,0 +1,192 @@
+    private final OpenSslServerSessionContext sessionContext;
+    private final OpenSslKeyMaterialManager keyMaterialManager;
+
+    ReleasableOpenSslServerContext(
+            X509Certificate[] trustCertCollection, TrustManagerFactory trustManagerFactory,
+            X509Certificate[] keyCertChain, PrivateKey key, String keyPassword, KeyManagerFactory keyManagerFactory,
+            Iterable<String> ciphers, CipherSuiteFilter cipherFilter, ApplicationProtocolConfig apn,
+            long sessionCacheSize, long sessionTimeout, ClientAuth clientAuth) throws SSLException {
+        this(trustCertCollection, trustManagerFactory, keyCertChain, key, keyPassword, keyManagerFactory, ciphers,
+                cipherFilter, toNegotiator(apn), sessionCacheSize, sessionTimeout, clientAuth);
+    }
+
+    private ReleasableOpenSslServerContext(
+            X509Certificate[] trustCertCollection, TrustManagerFactory trustManagerFactory,
+            X509Certificate[] keyCertChain, PrivateKey key, String keyPassword, KeyManagerFactory keyManagerFactory,
+            Iterable<String> ciphers, CipherSuiteFilter cipherFilter, OpenSslApplicationProtocolNegotiator apn,
+            long sessionCacheSize, long sessionTimeout, ClientAuth clientAuth) throws SSLException {
+        super(ciphers, cipherFilter, apn, sessionCacheSize, sessionTimeout, SSL.SSL_MODE_SERVER, keyCertChain,
+              clientAuth, true);
+        // Create a new SSL_CTX and configure it.
+        boolean success = false;
+        try {
+            ServerContext context = newSessionContext(this, ctx, engineMap, trustCertCollection, trustManagerFactory,
+                                                      keyCertChain, key, keyPassword, keyManagerFactory);
+            sessionContext = context.sessionContext;
+            keyMaterialManager = context.keyMaterialManager;
+            success = true;
+        } finally {
+            if (!success) {
+                release();
+            }
+        }
+    }
+
+    @Override
+    public OpenSslServerSessionContext sessionContext() {
+        return sessionContext;
+    }
+
+    @Override
+    OpenSslKeyMaterialManager keyMaterialManager() {
+        return keyMaterialManager;
+    }
+
+    static final class ServerContext {
+        OpenSslServerSessionContext sessionContext;
+        OpenSslKeyMaterialManager keyMaterialManager;
+    }
+
+    static ServerContext newSessionContext(ReleasableOpenSslContext thiz, long ctx, OpenSslEngineMap engineMap,
+                                           X509Certificate[] trustCertCollection,
+                                           TrustManagerFactory trustManagerFactory,
+                                           X509Certificate[] keyCertChain, PrivateKey key,
+                                           String keyPassword, KeyManagerFactory keyManagerFactory)
+            throws SSLException {
+        checkNotNull(keyCertChain, "keyCertChainFile");
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Nullcheck of keyCertChain at line 110 of value previously dereferenced in io.netty.handler.ssl.ReleasableOpenSslServerContext.newSessionContext(ReleasableOpenSslContext, long, OpenSslEngineMap, X509Certificate[], TrustManagerFactory, X509Certificate[], PrivateKey, String, KeyManagerFactory) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE)